### PR TITLE
feat(ProfileBaner): if name too long, truncated and capacity displayed

### PR DESCRIPTION
**Pull request by ASparton**

### CODE REVIEW

File: front/src/components/Search/Result/ResultProfileDrawer.tsx [hunk 1]
```diff
@@ -27,7 +28,7 @@ const ResultProfileDrawer: React.FC<IResultProfileDrawerProps> = ({
         resultProfile!.id,
         EProfileType.HOST === profileType
       ).then((res) => {
-        setProfile(res?.data);
+        setProfile(buildProfile(res?.data as IHostProfile) as IHostProfile);
```
Comment (filveith) [inline front/src/components/Search/Result/ResultProfileDrawer.tsx:31]: Is it me or is buildProfile() not taking any parameters ?

Comment (filveith) [reply front/src/components/Search/Result/ResultProfileDrawer.tsx:31]: My bad, GitHub was redirecting to a method with the same name

Comment (filveith) [reply front/src/components/Search/Result/ResultProfileDrawer.tsx:31]: But why do we cast res.data to IHostPtofile when buildProfile takes *any*. I think isntad of *any* you should have IHostProfile | IArtistProfile, so no need to cast it afterwards

File: front/src/types/IHostProfile.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import ICapacity from './ICapacity';
 import IProfile from './IProfile';
 
 export default interface IHostProfile extends IProfile {
   capacityId?: number;
+  capacity: ICapacity;
```
Comment (filveith) [inline front/src/types/IHostProfile.ts:6]: ICapacity already has the capacityId so you can remove capacityId

File: front/src/components/ProfileView/ProfileBanner.tsx [hunk 3]
```diff
@@ -30,13 +39,18 @@ const ProfileBanner: React.FC<IProfileBannerProps> = ({
...
 }) => {
   const { t } = useTranslation();
   const { editMode, editConfirmed } = useProfileEditMode();
   const canEdit = useLocation().pathname.includes('/auth/profile');
-  const genresToDisplay = loading ? loadingGenres : genres;
+  let genresToDisplay = loading ? loadingGenres : genres;
+  genresToDisplay =
+    genresToDisplay.length > 2
+      ? [genresToDisplay[0], genresToDisplay[1]]
+      : genresToDisplay;
```
Comment (filveith) [inline front/src/components/ProfileView/ProfileBanner.tsx:53]: Is this correctly updating the component when you re-assign genresToDisplay ?

**Review by filveith — APPROVED** [review]

File: front/src/components/ProfileView/ProfileBanner.tsx (hunk 3 above)
Comment (ASparton) [reply front/src/components/ProfileView/ProfileBanner.tsx:53]: Yup, tested this morning

File: front/src/components/ProfileView/Gallery.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ const Gallery: React.FC<IGalleryProps> = ({
           media.type === EMediaType.IMAGE && (
             <Carousel.Slide key={media.id}>
               <Image
-                src={GigzFetcher.getImageUri(media.source)}
+                src={GigzFetcher.getImageUri(media.media)}
```
Comment (filveith) [inline front/src/components/ProfileView/Gallery.tsx:31]: :vomiting_face:

File: front/src/pages/Profile/Profile.tsx [hunk 5]
```diff
@@ -57,8 +36,12 @@ const Profile: React.FC = () => {
   useEffect(() => {
     getProfile(user.getProfileType() as EProfileType).then((profile) => {
       setProfileType(user.getProfileType() as EProfileType);
+      console.log(profile.data);
```
Comment (filveith) [inline front/src/pages/Profile/Profile.tsx:39]: :eyes:

**Review by filveith — APPROVED** [review]

**Review by filveith — APPROVED** [review]

**Pull request merged**